kenny-chalupa/cdf-fabric-replicator
# 35730 Add Time Series to Extractor

Added Functionality for Extracting Time Series Data from Lakehouse into CDF by adding logic and methods into the extractor class. These changes should highlight the following:

- Time Series Extraction from Lakehouse to CDF in Extractor Method
- Logic to handle checkpointing in the state store by time-series timestamp
- Refactored code to help split logic between Extracting Events and Extracting Time-Series  

## How to Verify 

-  in `example_config.py`:
  -  change source. event_path  to: Tables/TestEvents
  - source.abfss_prefix to: abfss://ebbedac0-b667-4db5-895f-86eba7a8bc44@msit-onelake.dfs.fabric.microsoft.com/8e66858c-c1f3-40b6-92e3-a86e747a7c9d
  - source.file_path to: Files/tank_thumbnail.png
  -  source.abfss_raw_time_series_path to: Tables/mtu_historic
  - source.data_set_id to: 3351080165451477

  - set destination.time_series_prefix to your desired unique time series name prefix: current one is "ktst_new_extract_" 
-  Run the project with the command in the terminal `poetry run cdf_fabric_replicator example_config.yaml `
- After running you should double check the CDF portal to see the time-series data points being populated:

![image](https://github.com/kenny-chalupa/cdf-fabric-replicator/assets/50754008/aba99319-b5fe-4a29-8c5f-3e52435b94b1)
